### PR TITLE
Check Rust spelling in CI

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
Tooling for checking for spelling mistakes has been in the `nns-dapp` repository for some weeks now.  The checks for spelling mistakes in `rustdoc` have been consistently near zero and the tooling seems stable, so it is ripe for use in CI.

# Changes
- Add `cargo spellcheck` to CI.

# Tests
- See CI

# Todos

- [x] Add entry to changelog (if necessary).

### CODE REVIEW

File: rs/backend/src/accounts_store/schema.rs [hunk 1]
```diff
@@ -97,7 +97,7 @@ pub trait AccountsDbTrait {
 /// Note: The numeric representations of these labels are guaranteed to be stable.
 #[repr(u32)]
 pub enum SchemaLabel {
-    /// Data is stored on the heap in a BTreeMap and serialized to stable memory on upgrade.
+    /// Data is stored on the heap in a `BTreeMap` and serialized to stable memory on upgrade.
```
Comment (bitdivine) [inline rs/backend/src/accounts_store/schema.rs:100]: The spellchecker complains when code isn't  quoted.

File: .github/workflows/checks.yml [hunk 2]
```diff
@@ -33,6 +33,19 @@ jobs:
...
                   exit 1
           }
+  spelling:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install spellcheck
+        run: |
+          cd /
```
Comment (nmattia) [inline .github/workflows/checks.yml:43]: nit: why move to the root?

File: .github/workflows/checks.yml [hunk 3]
```diff
@@ -33,6 +33,19 @@ jobs:
...
+          sudo apt-get update
+          sudo apt-get install -yy hunspell hunspell-en-gb hunspell-en-us clang libclang-dev
+          cargo install --target x86_64-unknown-linux-gnu cargo-spellcheck
```
Comment (nmattia) [inline .github/workflows/checks.yml:46]: nit: isn't the target the default? this seems to make the code less portable for no apparent reason

**Review by nmattia — APPROVED** [review]
LGTM, thanks!

File: .github/workflows/checks.yml (hunk 3 above)
Comment (bitdivine) [reply .github/workflows/checks.yml:46]: It defaults to `musl` which is normally a fine choice but in this particular case it fails to compile.

**Comment by bitdivine** [comment]
Could you re-approve, please, @nmattia ?  I had to merge in main and fix more spelling and under the merge rules we are trialling, that requires a fresh review.

**Review by nmattia — APPROVED** [review]

**Pull request merged**